src/cmap/command_monitoring_events.ts

```diff
@@ -181,9 +181,8 @@ const HELLO_COMMANDS = new Set(['hello', LEGACY_HELLO_COMMAND, LEGACY_HELLO_COMM
 
 // helper methods
 const extractCommandName = (commandDoc: Document) => Object.keys(commandDoc)[0];
-const namespace = (command: WriteProtocolMessageType) => command.ns;
-const databaseName = (command: WriteProtocolMessageType) => command.ns.split('.')[0];
-const collectionName = (command: WriteProtocolMessageType) => command.ns.split('.')[1];
+const namespace = (command: Query) => command.ns;
+const collectionName = (command: Query) => command.ns.split('.')[1];
```

Comment:
Isn't this always just `$cmd`?

```ts
    const ns = `${databaseName}.$cmd`;
```

Reply:
Not quite because Query instances are never command monitored, this helper is never used. We have a lot of Query-handling logic that we can delete here but I'd say its worth another PR for that.